cms/jinja2/templates/pages/methodology_page.html

```diff
@@ -10,44 +10,48 @@
 {%- set breadcrumbs_label = _("Breadcrumbs") -%}
 
 {% block header_area %}
-    {% if page.contact_details %}
-        {% set contact_description_html -%}
-            <a href="mailto:{{ page.contact_details.email }}">{{ page.contact_details.name }}</a>
-        {%- endset %}
+    {# fmt:off #}
+    {% set publication_item = {
+        "term": published_label,
+        "descriptions": [
+            {
+                "description": page.publication_date|date("DATE_FORMAT")
+            }
+        ]
+    } %}
 
-        {# fmt:off #}
-        {% set contact_details =
+    {% set last_revised_item = page.last_revised_date and {
+        "term": last_revised_label,
+        "descriptions": [
             {
-                "term": "Contact:",
-                "descriptions": [
-                    {
-                        "description": contact_description_html
-                    }
-                ]
+                "description": page.last_revised_date|date("DATE_FORMAT")
             }
-        %}
-        {# fmt:on #}
-    {% else %}
-        {% set contact_details = "" %}
-    {% endif %}
+        ]
+    } or None %}
+
+    {% set contact_description_html -%}
+        <a href="mailto:{{ page.contact_details.email }}">{{ page.contact_details.name }}</a>
+    {%- endset %}
 
-    {% if page.last_revised_date %}
-        {# fmt:off #}
-        {% set last_revised =
+    {% set contact_item = page.contact_details and {
+        "term": _("Contact:"),
+        "descriptions": [
             {
-                "term": last_revised_label,
-                "descriptions": [
-                    {
-                        "description": page.last_revised_date|date("DATE_FORMAT")
-                    }
-                ]
+                "description": contact_description_html
             }
-        %}
-        {# fmt:on #}
-    {% else %}
-        {% set last_revised = "" %}
-    {% endif %}
+        ]
+    } or None %}
+    {# fmt:on #}
 
+    {# filter out any optional items which haven't been set #}
+    {# fmt:off #}
+    {% set itemsList = [
+        publication_item,
+        last_revised_item,
+        contact_item
+    ] | selectattr('term') | list %}
+    {# fmt:on #}
```

Comment:
thought:


```
"itemsList": [publication_item, last_revised_item, contact_item] | selectattr('term') | list,
```

directly in the component call.

Same on the stats article page

Reply:
Nice suggestion, much prefer this thank you! ⭐ Updated